src/ATen/native/xpu/mkl/BatchLinearAlgebra.cpp

```diff
@@ -0,0 +1,1243 @@
+        LU_pivots_working_copy,
+        infos,
+        TransposeType::NoTranspose);
+  });
+
+  std::copy(
+      infos.begin(), infos.end(), infos_tensor.template data_ptr<int32_t>());
+  at::_linalg_check_errors(infos_tensor, "lu_solve_xpu", self.dim() == 2);
+
+  return self_working_copy;
+}
+
+Tensor _det_lu_based_helper_backward_helper(
+    const Tensor& det_grad,
+    const Tensor& det,
+    const Tensor& self,
+    const Tensor& lu,
+    const Tensor& pivs) {
+  auto eps = at::native::_get_epsilon(c10::toRealValueType(self.scalar_type()));
+  auto n = self.size(-1);
+  auto eps_tensor = at::tensor(eps, self.options());
+  auto condition_diagonal = [&](const Tensor& x) {
+    auto x_diag = x.diagonal(0, -2, -1);
+    auto x_diag_conditioned = at::where(x_diag == 0.0, eps_tensor, x_diag);
+    x_diag.copy_(x_diag_conditioned);
+  };
+
+  // create a matrix d := (det_grad * det.conj()) I
+  // NOTE: we do not use the shorter version
+  // auto d = at::zeros_like(self);
+  // d.diagonal(0, -2, -1).copy_((det_grad * det.conj()).unsqueeze(-1));
+  // to avoid in-place operations to eliminate potential issues with Vmap
+  auto det_expanded_sizes = det.sizes().vec();
+  det_expanded_sizes.push_back(n);
+  auto d_diag = det_grad * det.conj();
+  auto d = at::diag_embed(d_diag.unsqueeze(-1).expand(det_expanded_sizes));
+  // make sure that d is Fortran-contiguous. The transposition is sufficient as
+  // d is a diagonal square matrix
+  d = d.transpose(-2, -1);
+
+  // we want to condition the diagonal of the lu Tensor, but it is not allowed
+  // to modify arguments of backward functions in-place, hence the cloning.
+  auto lu_clone = lu.clone();
+  condition_diagonal(lu_clone);
+
+  auto trans = self.is_complex() ? TransposeType::ConjTranspose
+                                 : TransposeType::Transpose;
+  auto infos_tensor = at::zeros(
+      native::batchCount(d),
+      self.options().dtype(kInt).device(DeviceType::CPU));
+  std::vector<int32_t> infos(native::batchCount(d), 0);
+
+  // d is modified in-place and will contain the result
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(
+      d.scalar_type(), "_det_lu_based_helper_backward_helper", [&] {
+        apply_lu_solve_xpu_<scalar_t>(d, lu_clone, pivs, infos, trans);
+      });
+
+  std::copy(
+      infos.begin(), infos.end(), infos_tensor.template data_ptr<int32_t>());
+  at::_linalg_check_errors(
+      infos_tensor, "_det_lu_based_helper_backward_helper", self.dim() == 2);
+
+  return d;
+}
+
+void lu_solve_mkl(
+    const Tensor& LU,
+    const Tensor& pivots,
+    const Tensor& B,
+    TransposeType trans) {
+  std::vector<int32_t> infos_vec(native::batchCount(LU), 0);
+
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(LU.scalar_type(), "lu_solve_xpu", [&] {
+    apply_lu_solve_xpu_<scalar_t>(B, LU, pivots, infos_vec, trans);
+  });
+}
+
+void lu_factor_mkl(
+    const Tensor& LU,
+    const Tensor& pivots,
+    const Tensor& info,
+    bool pivot) {
+  TORCH_CHECK(
```

Comment:
please align with stock pytorch `lu_factor_kernel`, logic, we don't need those create/copy/resize code here

Reply:
The create/copy of `pivots` cannot be remove due to inconsistent API between PyTorch and oneMKL. The other create/copy/resize code has been removed.